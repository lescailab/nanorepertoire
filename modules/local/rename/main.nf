process RENAME {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/fastx_toolkit:0.0.14--hdbdd923_11' :
        'biocontainers/fastx_toolkit:0.0.14--hdbdd923_11' }"

    input:
    tuple val(meta), path(reads)
    val single_end

    output:
    tuple val(meta), path('*.fastq.gz'), emit: renamed
    path 'versions.yml'               , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    if (single_end) {
        """
        zcat ${reads[0]} | fastx_renamer -z -n COUNT -o ${meta.id}_renamed.fastq.gz

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
        rename: \$(fastx_renamer -h |& sed '2!d' |& perl -nae 'print \$F[4]."\n";')
        END_VERSIONS
        """
    }
    else {
        """
        zcat ${reads[0]} | fastx_renamer -z -n COUNT -o ${meta.id}_R1_renamed.fastq.gz
        zcat ${reads[1]} | fastx_renamer -z -n COUNT -o ${meta.id}_R2_renamed.fastq.gz

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            rename: \$(fastx_renamer -h |& sed '2!d' |& perl -nae 'print \$F[4]."\n";')
        END_VERSIONS
    """
    }
}
