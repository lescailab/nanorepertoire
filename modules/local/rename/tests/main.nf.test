nextflow_process {
    name "Test Process RENAME"
    script "../main.nf"
    process "RENAME"
    config "./nextflow.config"
    tag "modules"
    tag "modules_nfcore"
    tag "rename"
    tag "renamingflash"

    test("rename-single_end") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test1', single_end:false ], // meta map
                    [file("https://github.com/lescailab/testdata/raw/master/nanobodies/test01_flash_merged.fastq.gz", checkIfExists: true)
                    ]
                ]
                input[1] = true
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }

}
