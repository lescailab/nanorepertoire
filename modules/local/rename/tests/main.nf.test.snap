{
    "rename-single_end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1_renamed.fastq.gz:md5,a881ed33b497636ecbb791513dc99055"
                    ]
                ],
                "1": [
                    "versions.yml:md5,9624dbd934ca2f37588004000eda5cc7"
                ],
                "renamed": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1_renamed.fastq.gz:md5,a881ed33b497636ecbb791513dc99055"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9624dbd934ca2f37588004000eda5cc7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-15T10:47:23.664611"
    }
}
